main.go

```diff
@@ -1,16 +1,38 @@
 package main
 
 import (
-	"fmt"
+	"encoding/json"
 	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
 )
 
-func main() {
-	http.HandleFunc("/hello", HelloHandler)
-	fmt.Println("running server ")
-	http.ListenAndServe(":8080", nil)
+type Request struct {
+	Name string `json:"name"`
+}
+
+type Response struct {
+	Code      int    `json:"code"`
+	Message   string `json:"message"`
+	Timestamp string `json:"timestamp"`
 }
 
 func HelloHandler(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "Hello, %s!\n", r.URL.Path[1:])
+	w.Header().Set("Content-Type", "application/json")
+	var req Request
+	json.NewDecoder(r.Body).Decode(&req)
+	defer r.Body.Close()
+	response := &Response{
+		Code:      200,
+		Message:   "Welcome " + req.Name + "!",
+		Timestamp: time.Now().UTC().Format(time.RFC3339),
+	}
+	json.NewEncoder(w).Encode(response)
+}
```

Comment:
This code is in the other file too. Then why here?

Reply:
cleared the confusion